app/[addressOrDomain]/page.tsx

```diff
@@ -435,16 +528,45 @@ export default function Page({ params }: AddressOrDomainProps) {
       );
       console.log("Error while calculating address portfolio stats", error);
     }
+  }, [address]);
 
-    setLoadingProtocols(false);
-  }, []);
+  const fetchPortfolioData = useCallback(async (addr: string) => {
+    setLoadingProtocols(true);
+    try {
+      // Argent API requires lowercase address
+      const normalizedAddr = addr.toLowerCase();
+      const [dappsData, tokensData, userTokensData, userDappsData] =
+        await Promise.all([
+          fetchDapps(),
+          fetchTokens(),
+          fetchUserTokens(normalizedAddr),
+          fetchUserDapps(normalizedAddr),
+        ]);
+
+      const data = {
+        dapps: dappsData,
+        tokens: tokensData,
+        userTokens: userTokensData,
+        userDapps: userDappsData,
+      };
+
+      await Promise.all([
+        fetchPortfolioProtocols(data),
+        fetchPortfolioAssets(data),
+      ]);
+    } catch (error) {
+      showNotification("Error while fetching address portfolio", "error");
```

Comment:
Could you implement some retry mechanism for all requests to portfolio ? It looks like the first time we fetch any address, the request fails (and after like 10 seconds of retrying it finally works). I think this is because it takes some time for them to gather and cache all the data

Reply:
Yep I'll do an API wrapper with retry feature now, sorry for the delay